src/main/java/view/SearchView.java

```diff
@@ -112,6 +116,46 @@ public void actionPerformed(ActionEvent e) {
                 SearchView.this.addFilter("Cast", castField.getText());
             }
         });
+
+        // Action listener for keyword search button (search by name)
+        keywordSearchButton.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                final String keyword = keywordField.getText().trim();
```

Comment:
Aren't you calling trim in the interactor as well? I'd say you could get rid of it here, or do it in the controller.